ledger-core/v2/virtual/handlers/vstatereport.go

```diff
@@ -45,12 +47,32 @@ func (s *SMVStateReport) Init(ctx smachine.InitializationContext) smachine.State
 
 func (s *SMVStateReport) stepProcess(ctx smachine.ExecutionContext) smachine.StateUpdate {
 	catalog := object.Catalog{}
-	if s.Payload.ProvidedContent == nil || s.Payload.ProvidedContent.LatestDirtyState == nil {
-		panic(throw.IllegalValue())
+
+	objectRef := s.Payload.Callee
+	immutablePendingCount := int(s.Payload.ImmutablePendingCount)
+	mutablePendingCount := int(s.Payload.MutablePendingCount)
+
+	var (
+		objectDescriptor descriptor.Object
+
+		immutableExecuteSemaphore smsync.SemaphoreLink
+		mutableExecuteSemaphore   smsync.SemaphoreLink
+
+		deactivated *bool
+	)
+
+	if s.Payload.ProvidedContent != nil && s.Payload.ProvidedContent.LatestDirtyState != nil {
+		objectDescriptor = descriptor.NewObject(
+			objectRef,
+			s.Payload.ProvidedContent.LatestDirtyState.Reference,
```

Comment:
I would assign 's.Payload.ProvidedContent.LatestDirtyState' to some variable and would use this variable